crates/evm/src/execute.rs

```diff
@@ -161,6 +162,152 @@ pub trait BlockExecutorProvider: Send + Sync + Clone + Unpin + 'static {
         DB: Database<Error: Into<ProviderError> + Display>;
 }
 
+/// A factory for creating block execution strategies.
+pub trait BlockExecutionStrategyFactory<DB>
+where
+    DB: Database,
+{
+    /// The specific [`BlockExecutionStrategy`] type this factory produces.
+    type Strategy: BlockExecutionStrategy<DB>;
+
+    /// The error type returned by this factory and its produced strategies.
+    type Error: From<ProviderError>
+        + From<<Self::Strategy as BlockExecutionStrategy<DB>>::Error>
+        + core::error::Error;
+
+    /// Creates a new block execution strategy instance.
+    fn create(
+        &self,
+        block: &BlockWithSenders,
+        total_difficulty: U256,
+    ) -> Result<Self::Strategy, Self::Error>;
+}
+
+/// Defines the strategy for executing a single block.
+pub trait BlockExecutionStrategy<DB> {
+    /// The error type returned by this strategy's methods.
+    type Error: From<ProviderError> + core::error::Error;
+
+    /// Configures the environment before start applying any changes.
+    fn configure_environment(&mut self) -> Result<(), Self::Error>;
```

Comment:
perhaps this should actually be part of the create step? because we already have the block as arg there

Reply:
makes sense, removed